pkg/session/bench_test.go

```diff
@@ -1944,3 +1945,65 @@ func TestBenchDaily(t *testing.T) {
 		BenchmarkAutoIncrement,
 	)
 }
+
+func BenchmarkPipelinedInsertIgnore(b *testing.B) {
+	se, do, st := prepareBenchSession()
+	defer func() {
+		se.Close()
+		do.Close()
+		st.Close()
+	}()
+	mustExecute(se, `create table tmp (id int, dt varchar(512))`)
+	mustExecute(se, `create table src (id int, dt varchar(512))`)
+	for i := 0; i < 100; i++ {
+		mustExecute(se, "begin")
+		for lines := 0; lines < 100; lines++ {
+			mustExecute(se, "insert into src values (42, repeat('x', 512)), (66, repeat('x', 512))")
+		}
+		mustExecute(se, "commit")
+	}
+
+	b.ResetTimer()
+	se.GetSessionVars().BulkDMLEnabled = true
+	se.GetSessionVars().StmtCtx.InInsertStmt = true
+	for i := 0; i < b.N; i++ {
+		se.Execute(context.Background(), "insert ignore into tmp select * from src")
+		warningCount := se.GetSessionVars().StmtCtx.WarningCount()
+		require.Zero(b, warningCount)
+	}
+	b.StopTimer()
+}
+
+func BenchmarkPipelinedInsertOnDuplicate(b *testing.B) {
```

Comment:
Can you add this benchmark to dailybench?

Reply:
I didn't do this because I don't know how it's maintained. @tiancaiamao how do you like it?